### DISCUSSION TITLE
Add pywal to generate colorschemes based on wallpaper!!

### DISCUSSION THREAD

**Discussion by H1manshu-Soni** [prasanthrangan/hyprdots#872] [Ideas]
I have create a fork and made required changes (just added three lines) I wanted to ask here before creating a pull request if you guys like the idea it can be used to rice firefox and sublime-text even telegram.
If you guys like the idea then let me know.

**Comment by kRHYME7** [comment]
Yeah, nice!

However, we have this so-called wallbash here, which themes waybar, Dunst, rofi.  in my case kitty and swaylock,  swaync, cava. 
https://github.com/prasanthrangan/hyprdots/discussions/811

And also I guess there is a way to add this so that not all users need to have pywal installed.  Please refer to the link I shared. with that you can add external scripts and be run whenever you change your wallpaper.


This will help make the code modular 

I will help: 

The first file will be the control file it tells the wallpaper script what to run  ``` ~/.config/hypr/wallbash/pywal.dcol ```
```
/tmp/pywal |${ScrDir}/wallbashpywal.sh
```
this file has 3 parts. 1st column | 2nd column and the text below it. We will only need the 2nd column.   
We don't need the 1st column so we can just redirect it somewhere
That second column is the script the wallpaper script need to run . 

now the script should have there requirements e.g
* It handles if the dependencies are installed. 
     in our case we have ``` python-pywal ``` and ``` python-pywalfox ``` as dependency 

* Also it handles file or the script  permission 
    e.g 
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/7a502ecd-44c8-4e72-82f0-ecccb34d34ea)
This one needs to ``` pywalfox install ``` also  needs to handle the permission

* The script can be called multiple times and should exit

**Reply by H1manshu-Soni** [reply]
but users must have pywalfox extension in firefox installed otherwise it wont work :(

**Reply by prasanthrangan** [reply]
Hey, I wont be integrating pywal here, as I've no idea how/what colors it extract from wallpaper and it wont be the same colors as generated by wallbash.
If you are trying to theme an application I'd implement it directly with wallbash.

**Reply by H1manshu-Soni** [reply]
No one has implemented anything chill 
it is still just an idea if you don't like it sure no problem :+1:

**Reply by prasanthrangan** [reply]
No worries bro, please let me know which applications you are trying to theme, just firefox?
I'll look into integrating it with wallbash!

Also @kRHYME7, could you please share dcol files for kitty and swaylock, swaync, cava you have created, we can add that in here as well.

**Reply by kRHYME7** [reply]
@prasanthrangan Sure bro.  I'll clean this up then make a MR soon.

**Reply by RWayne93** [reply]
> @prasanthrangan Sure bro.  I'll clean this up then make a MR soon. 
> 
> 
> 
> 
> 
> 

How do you get applications to switch colors based on the selected theme. This is all being done by wallbash at least that's my assumption after reading this thread? There's a few applications I've been using that I would love to switch when I switch Themes. Right now I have just been adjusting the colors manually.

**Reply by kRHYME7** [reply]
@RWayne93 Here's a recent documentation.

https://github.com/prasanthrangan/hyprdots/wiki/Wallbash

Feel free to clarify something.

**Comment by TNitroo** [comment]
yeah, I would definitely love to have such function in hyperdots, I just downloaded it yesterday, so I still have no idea how I could do anything like this myself, so I would love this if this was an selectable option by default or something like that.

**Reply by kRHYME7** [reply]
``` SUPER + SHIFT + D ```  Try 


https://github.com/prasanthrangan/hyprdots/discussions/872?notification_referrer_id=NT_kwDOAy8V47M5NDUwMjE5Mjg3OjUzNDE3NDQz#discussioncomment-8455662

**Reply by TNitroo** [reply]
nvm, my bad, thanks for telling me about this! I love hyperdots even more now :yaay:

**Reply by kRHYME7** [reply]
If terminal contrast is not good, you can add this command to your key bind ``` Hyprdots wallbash term -t ```

This is on testing, so I only added 3 combinations to alter terminal color.

**Reply by TNitroo** [reply]
I really appreciate it man! many thanks to you! :yatta:
